### ISSUE TITLE
[ENH]: Use a config.sh file

### ISSUE THREAD

**Issue by idr4n** [2KAbhishek/tdo#9]
### Description

Hi there,

I have been using tdo in my daily workflow, and as mentioned before, I really like it for its simplicity and aligned to what I was doing before.

I have done a few modifications to adjust it better to my own workflow, such as not adding a timestamp every time I use `tdo e`, only when creating new notes, and modified the timestamp in that case (also, my note.md template is completely empty). These are the minor changes so far:

```sh
new_note() {
    root="$NOTES_DIR"
    note_file="$root/notes/$1.md"
    template="$root/templates/note.md"
    create_file "$note_file" "$template"
->    # add file name as the note title
->    echo -e "# $1" >>"$note_file"
->    add_timestamp "$note_file" "Created: $(date +'%a. %b %d, %Y at %I:%M%p')"
    write_file "$note_file" "$root"
}

new_entry() {
    root="${JOURNAL_DIR:-$NOTES_DIR}"
    entry_file="$root/entries/$(generate_file_path "$1")"
    template="$root/templates/entry.md"
    create_file "$entry_file" "$template"
->    # add_timestamp "$entry_file"
    write_file "$entry_file" "$root"
}
```

Obviously this is for my own preference, so I was thinking whether it would be a good idea to have a `config.sh` file and source it in `tdo.sh` with some config variables, such as whether to add timestamps on entry notes, new notes, etc., and what time of timestamp and note title to add.

**Comment by 2KAbhishek** [comment]
That sounds good.

Why don't you open up a PR for this, and instead of having a config.sh file let's load things from env.

It'll be great if you can add the config options on the readme file as well, absence of any of the configs should make the script use the current default values.

**Comment by idr4n** [comment]
I can see that using env variables is somehow convenient but was wondering whether you prefer to use env over a config file due to the integration with the nvim plugin, or for any other particular reason. Personally I prefer not to define many env variables in my shell config files but that's just my own preference. Would you consider having both options available, that is, checking for both env and variables defined in an config.sh file so the user can set either?

**Comment by 2KAbhishek** [comment]
Sure, let's have both, but the config file can't be named config.sh, it'll be very generic.

Some notes for implementation:

Let's call it tdorc, we'll also have to put it in the correct location ~/.config/tdorc should do initially.

In terms of the config file structure it can just be value assignments in sh, e.g

```bash
ADD_TIMESTAMP=false
```

In terms of priority, env values should take precedence over tdorc

Let me know if you have any other queries, else I can do a review when you have the PR up

**Comment by idr4n** [comment]
˙Sounds good! I will ask you along the way if needed. Thanks again!

**Comment by idr4n** [comment]
@2KAbhishek I will be opening up a PR soon, just to let you know. I'm doubling checking/testing as I have implemented the two options, that is, reading from env if set, or sourcing a config file. Is `~/.config/tdo/tdorc` an OK location?

Also, the variables that could be set are:

```sh
TIMESTAMP_ENTRY=false
TIMESTAMP_NEWNOTE=false
FILE_NAME_AS_TITLE=false # only apply on new notes, not if created already
ENTRY_TIMESTAMP="2024-01-01"
NOTE_TIMESTAMP="Created: $(date +'%a. %b %d, %Y at %I:%M%p')" # only apply on new notes, not if created already
```

All of them have defaults based on the current setting, so the behavior should be the same as it is right now if not defined as env or in `tdorc`.

I will work on the README once you review the initial PR.

**Comment by 2KAbhishek** [comment]
@idr4n Ya, having the default behaviour not change will definitely be good.

For the file location, I think we can skip the folder, just `~/.config/tdorc` should do

Will share more thoughts once the PR is up :)

**Comment by 2KAbhishek** [comment]
resolved with #10
